### PR TITLE
Batch backports 6.0.x/1.1: Backports of 5550, 5604

### PR DESCRIPTION
**Pull request by jlucovsky**
Continuation of #8975 

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
- [5604](https://redmine.openinfosecfoundation.org/issues/5604)
- [5550](https://redmine.openinfosecfoundation.org/issues/5550)

Describe changes:
- Backport of commits for 5604, 5550.

Updates:
- Abandon StreamSlice changes in dns.rs per review comment.

### Provide values to any of the below to override the defaults.

To use a pull request use a branch name like `pr/N` where `N` is the pull request number.

```
SV_REPO=
SV_BRANCH=
SU_REPO=
SU_BRANCH=
LIBHTP_REPO=
LIBHTP_BRANCH=
```

### CODE REVIEW

File: rust/src/applayer.rs [hunk 1]
```diff
@@ -23,6 +23,7 @@ use crate::filecontainer::FileContainer;
 use crate::applayer;
 use std::os::raw::{c_void,c_char,c_int};
 
+#[repr(C)]
 #[repr(C)]
```
Comment (jasonish) [inline rust/src/applayer.rs:27]: Duplicate line now.

**Comment by jasonish** [comment]
Otherwise DNS appears to be OK.  The SV test `dns/dns-invalid-opcode` should get updated to be not so 7.0 specific I guess.. It currently also looks for fields that don't exist in 6.0.

**Comment by suricata-qa** [comment]
WARNING:
 
 | field | baseline | test | % |
|--|--|--|--| 
| | SURI_TLPR1_stats_chk | | |
| .tcp.synack | 6809615 | 6047299 | 88.81% |
| .tcp.rst | 4872947 | 4600901 | 94.42% | 

Pipeline 14297

**Comment by ct0br0** [comment]
Have a QA rebase ready for when this merges

File: rust/src/applayer.rs (hunk 1 above)
Comment (victorjulien) [reply rust/src/applayer.rs:27]: Fixing up during staging.

**Comment by victorjulien** [comment]
Merged in #8995, thanks!

**Pull request closed without merging**